backend/src/lang/employee.service.ts

```diff
@@ -0,0 +1,53 @@
+import { DialogFlowService } from './dialog-flow.service';
+import { EmploymentContractService } from './../database/employmentContract/employmentContract.service';
+
+export default class EmployeeService {
+
+    fieldsString = 'fields';
+    structValueString = 'structValue';
+    nameString = 'name';
+    stringValueString = 'stringValue';
+
+    constructor(private dialogFlowService: DialogFlowService,
+                private employmentContractService: EmploymentContractService) {}
+
+    /**
+     * Processes the response of a users input in the context of an employment contract
+     * @param {DetectIntentResponse} detectintent - The response received from dialogflow
+     */
+    public processEmployeeContract(detectIntent: DetectIntentResponse) {
+        const responseAction = this.dialogFlowService.extractAction(detectIntent);
+        const responseParam = this.dialogFlowService.extractParameter(detectIntent)[this.fieldsString][responseAction];
```

Comment:
Das müsste du zumindest noch etwas aufsplitten oder die types erweitern! Schaut irgendwie nicht sonderlich sauber aus ...

Reply:
was meinst du aufsplitten? was möchtest du hier aufgesplittet haben

Reply:
Die Ketten von Funktionsaufruf und Array-Zugriff empfinde ich als sehr schwer leserlich. (Es gab sie noch an anderer Stelle in deinem Code). Also in meiner Position fällt es schwer nachzuvollziehen was genau du in dieser Zeile machst.
Deutlich besser wäre es, wenn du die zwischen Ergebnisse in Variablen mit aussagekräftigem Namen speicherst. So kann jeder aus unserem Team ohne langes nachforschen nachvollziehen, was du genau machst.

Reply:
Sowas in der Art: 
const responseParam = this.dialogFlowService.extractParameter(detectIntent);
const actionOfDateParam = repsonseParam[this.fieldsString][responseAction];

So wird direkt aus dem Variablen Namen deutlich, welche Information du extrahiert hast. (Bei noch mehr Array-Zugriffen würde ich weitere Variablen einführen)